### DISCUSSION TITLE
Can Xray Team Improve XHTTP(SplitHTTP) Upload speed on ADSL?

### DISCUSSION THREAD

**Discussion by deepsm0ke** [XTLS/Xray-core#4029] [General]
Hello. 

So, the problem that has been and still is that the upload speed and stability on ADSL is not good at all (except for Annex M mode, which has relatively good adsl's upload about 3mbps). On the other hand, other transports like Websocket or httpUpgrade Over CDN have good upload on ADSL (It uploads slowly, but the important thing is that it starts uploading, whereas with splithttp it doesn't even start uploading.), but in XHTTP, we may have to wait (even in h3 alpn mode) for nearly 2 minutes (Maybe or maybe not) until only 128kb of that file is uploaded!. Any ideas or suggestions for improvement?.

Thanks.

**Comment by RPRX** [comment]
Try https://github.com/XTLS/Xray-core/commit/2d7b0e8cd4f1fc6ec840891dbc03b4e781d0a5ad

**Reply by deepsm0ke** [reply]
The problem was not resolved and still persists.

**Reply by RPRX** [reply]
> The problem was not resolved and still persists.

开一个 issue

**Reply by deepsm0ke** [reply]
Ok.
